### PR TITLE
issue/126 Fixed completion restore on new session

### LINKED ISSUE #126: does not restore state correctly on a new session
If the parent is complete when the trickle model is created, the trickle button should be considered complete.

### PR DESCRIPTION
**Pull request by oliverfoster**
fixes #126 

### Fixed
* when trickle model is created it now checks to see if it should be complete or not

### Added
* check to see if the parent article/block is complete

### CODE REVIEW

**Review by tomgreenfield — APPROVED** [review]
:eye:

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by eleanor-heath — APPROVED** [review]

File: js/TrickleButtonModel.js [hunk 1]
```diff
@@ -55,6 +61,15 @@ export default class TrickleButtonModel extends ComponentModel {
     });
   }
 
+  /**
+   * @returns {boolean} true if the parent container is already complete
+   */
+  isStepComplete() {
+    const completionAttribute = getCompletionAttribute();
+    const isParentComplete = this.getParent().get(completionAttribute);
```
Comment (eleanor-heath) [inline js/TrickleButtonModel.js:69]: you could `return this.getParent().get(completionAttribute);` instead of declaring a const and returning the const.

Comment (oliverfoster) [reply js/TrickleButtonModel.js:69]: It's mainly just for debugging purposes that it's like this atm.
Easier to breakpoint.

**Pull request merged**